docs/howto/mocking_services.md

```diff
@@ -0,0 +1,273 @@
+  def overloaded(arg1: Long)                 : ZIO[R, Nothing, String]
+}
+
+object ExampleService {
+  object static             extends Method[Unit, String]
+  object zeroArgs           extends Method[Unit, Int]
+  object zeroArgsWithParens extends Method[Unit, Long]
+  object singleArg          extends Method[Int, String]
+  object multiArgs          extends Method[(Int, Long), String]
+  object multiParamLists    extends Method[(Int, Long), String]
+  object command            extends Method[Int, Unit]
+  object overloaded {
+    object _1 extends Method[Int, String]
+    object _2 extends Method[Long, String]
+  }
+}
+```
+
+> **Note:** we're using tuples to represent multiple argument methods, which follows with a limit to max 22 arguments, as is itself Scala limited.
+
+For overloaded methods we simply nest a list of numbered objects, each representing subsequent overloads.
+
+Next, we create the mockable implementation of the service:
+
+```
+implicit val mockable: Mockable[AccountObserver] = (mock: Mock) =>
+  new AccountObserver {
+    val accountObserver = new AccountObserver.Service[Any] {
+      def processEvent(event: AccountEvent): UIO[Unit] = mock(AccountObserver.Service.processEvent, event)
+    }
+  }
+```
+
+> **Note:** To make our mockable implementation automatically discovered, we need to place it inside `AccountObserver` modules companion object.
+
+## Scrapping the boilerplate
+
+```scala mdoc:invisible
+trait ProductEvent
+```
+
+All of this machinery is repetitive and boring work, prone to simple mistakes. Using the `@Accessable` and `@Mockable` macros provided by
+[zio-macros][link-zio-macros] we get the _capability tags_, _access helper_ and _mockable implementation_ autogenerated for us:
+
+```scala mdoc:silent
+import zio.macros.access.Accessable
+import zio.macros.mock.Mockable
+import zio.console.Console
+
+@Accessable
+@Mockable
+trait ProductObserver {
+  val productObserver: ProductObserver.Service[Any]
+}
+
+object ProductObserver {
+  trait Service[R] {
+    def processEvent(event: ProductEvent): ZIO[R, Nothing, Unit]
+  }
+
+  // autogenerated `object Service { ... }`
+  // autogenerated `object > extends Service[ProductObserver] { ... }`
+  // autogenerated `implicit val mockable: Mockable[ProductObserver] = ...`
+
+  trait Live extends ProductObserver {
+    // dependency on Console module
+    val console: Console.Service[Any]
+
+    val productObserver = new Service[Any] {
+      def processEvent(event: ProductEvent): UIO[Unit] =
+        for {
+         _    <- console.putStrLn(s"Got $event")
+         line <- console.getStrLn.orDie
+         _    <- console.putStrLn(s"You entered: $line")
+        } yield ()
+    }
+  }
+}
+```
+
+## Setting up expectations
+
+Finally we're all set and can create ad-hoc mock environments with our services.
+
+```scala mdoc:silent
+import zio.test._
+import zio.test.Assertion._
+
+val event = new ProductEvent
+val app = ProductObserver.>.processEvent(event)
```

Comment:
Is it more clear to use `>` infix (e.g. `ProductObserver > processEvent(event)`)? I think I saw this a few other places so if you agree might be good to do a global find and replace

Reply:
Unfortunately as it is now, the infix notation does not work:
```scala
val app = ProductObserver > processEvent(event)
// raises compile error:
// ProductObserver.>.type does not take parameters
```

I'm not entirely sure what is going on there. Will research this later and (if possible) fix in another PR.